src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -627,6 +633,93 @@ public class DeleteValueCommand
         public string Name;
     }
 
+    public class EditVersioningCommand : UpdateDatabaseCommand
+    {
+        public string DatabaseName;
+        public VersioningConfiguration Configuration;
+
+        public void UpdateDatabaseRecord(DatabaseRecord databaseRecord)
+        {
+            databaseRecord.VersioningConfiguration = Configuration;
+        }
+
+        public EditVersioningCommand() : base(null)
+        {
+        }
+
+        public EditVersioningCommand(VersioningConfiguration configuration, string databaseName) : base(databaseName)
+        {
+            Configuration = configuration;
+        }
+
+        public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
+        {
+            record.VersioningConfiguration = Configuration;
+        }
+
+        public override void FillJson(DynamicJsonValue json)
+        {
+            json[nameof(VersioningConfiguration)] = TypeConverter.ToBlittableSupportedType(Configuration);
+        }
+    }
+
+    public class EditExpirationCommand : UpdateDatabaseCommand
+    {
+        public string DatabaseName;
+        public ExpirationConfiguration Configuration;
+        public void UpdateDatabaseRecord(DatabaseRecord databaseRecord)
+        {
+            databaseRecord.ExpirationConfiguration = Configuration;
+        }
+
+        public EditExpirationCommand() : base(null)
+        {
+        }
+
+        public EditExpirationCommand(ExpirationConfiguration configuration,string databaseName) : base(databaseName)
+        {
+            Configuration = configuration;
+        }
+
+        public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
+        {
+            record.ExpirationConfiguration = Configuration;
+        }
+
+        public override void FillJson(DynamicJsonValue json)
+        {
+            json[nameof(ExpirationConfiguration)] = TypeConverter.ToBlittableSupportedType(Configuration);
+        }
+    }
+
+    public class EditPeriodicBackupCommand : UpdateDatabaseCommand
```

Comment:
rename to PeriodicExport
move to ravendb/src/Raven.Server/ServerWide/Commands/PeriodicExport